### DISCUSSION TITLE
PDF Viewer does not allow for providing the PDF file directly

### DISCUSSION THREAD

**Discussion by Lucasharskamp** [vikramlearning/blazorbootstrap#628] [Q&A]
https://docs.blazorbootstrap.com/components/pdf-viewer 

For security purposes I don't want the PDF's to be accessible through an URL. It's all to be provided if the user has the correct permissions. Instead, I want the blazor component itself to provide the contents of the PDF file. As, say, a stream/byte array/base64 string/etc. Could such a thing be done?

**Comment by gvreddy04** [comment]
@Lucasharskamp Thank you for trying **BlazorBootstrap**. In the initial version, we added URL support only. Please create an issue based on this discussion. We will add support for streams, byte arrays, and base64 strings in the upcoming versions.

**Reply by smuuba** [reply]
I had success passing the file as a base64 string converted from the file-bytes, like so:

<PdfViewer Url="@($"data:application/pdf;base64,{Convert.ToBase64String(FileContents)}")" />

where FileContents is the byte array of the file.  Hope that helps.

**Reply by Lucasharskamp** [reply]
Thanks! That's indeed a good solution.

**Reply by gvreddy04** [reply]
@smuuba Thank you. I'll update the documentation and examples.
